docs/core/diagnostics/trace-perfcollect-lttng.md

```diff
@@ -0,0 +1,201 @@
+
+## Preparing Your Machine
+
+Follow these steps to prepare your machine to collect a performance trace with `perfcollect`.
+
+1. Download `perfcollect`.
+
+    > ```bash
+    > curl -OL http://aka.ms/perfcollect
+    > ```
+
+2. Make the script executable.
+
+    > ```bash
+    > chmod +x perfcollect
+    > ```
+
+3. Install tracing prerequisites - these are the actual tracing libraries.
+
+    > ```bash
+    > sudo ./perfcollect install
+    > ```
+
+    This will install the following prerequisites on your machine:
+
+    1. `perf`: the Linux Performance Events sub-system and companion user-mode collection/viewer application. `perf` is part of the Linux kernel source, but is not usually installed by default.
+
+    2. `LTTng`: Used to capture event data emitted at runtime by CoreCLR. This data is then used to analyze the behavior of various runtime components such as the GC, JIT and thread pool.
+
+## Collecting a Trace
+
+1. Have two shells available - one for controlling tracing, referred to as **[Trace]**, and one for running the application, referred to as **[App]**.
+
+2. **[App]** Setup the application shell with the following environment variables - this enables tracing configuration of CoreCLR.
+
+    > ```bash
+    > export COMPlus_PerfMapEnabled=1
+    > export COMPlus_EnableEventLog=1
+    > ```
+
+3. **[Trace]** Start collection.
+
+    > ```bash
+    > sudo ./perfcollect collect sampleTrace
+    > ```
+
+    Expected Output:
+
+    > ```bash
+    > Collection started.  Press CTRL+C to stop.
+    > ```
+
+4. **[App]** Run the app - let it run as long as you need to in order to capture the performance problem.
+
+    > ```bash
+    > dotnet run
+    > ```
+
+5. **[Trace]** Stop collection - hit CTRL+C.
+
+    > ```bash
+    > ^C
+    > ...STOPPED.
+    >
+    > Starting post-processing. This may take some time.
+    >
+    > Generating native image symbol files
+    > ...SKIPPED
+    > Saving native symbols
+    > ...FINISHED
+    > Exporting perf.data file
+    > ...FINISHED
+    > Compressing trace files
+    > ...FINISHED
+    > Cleaning up artifacts
+    > ...FINISHED
+    >
+    > Trace saved to sampleTrace.trace.zip
+    > ```
+
+    The compressed trace file is now stored in the current working directory.
+
+## Resolving Framework Symbols
+
+Framework symbols need to be manually generated at the time the trace is collected. They are different than app-level symbols because the framework is pre-compiled while apps are just-in-time-compiled. For code like the framework that was precompiled to native code, you need a special tool called crossgen that knows how to generate the mapping from the native code to the name of the methods.
```

Comment:
```suggestion
Framework symbols need to be manually generated at the time the trace is collected. They are different than app-level symbols because the framework is pre-compiled while app code is just-in-time-compiled. For framework code that was precompiled to native code, you need to call crossgen to generate the mapping from the native code to the name of the methods.
```